.travis.yml

```diff
@@ -17,25 +26,46 @@ services:
   - redis-server
 
 install:
-  - bash ./.travis/cc-download.sh
   # install local dependency repos first
-  - for x in  "lib/eslint-config-twine" "lib/twine-util"; do cd $x; npm install; cd ../..; done;
-  - for x in "api" "dashboard-app" "visitor-app"; do cd $x; npm install; cd ..; done;
+  - pip install --user awscli
+  - for f in lib/*; do cd $f; npm install; cd ../..; done;
+  - cd $TRAVIS_BUILD_DIR/$APP_DIR && npm install
 
 before_script:
+  - bash $TRAVIS_BUILD_DIR/.travis/cc_download.sh
   - psql -U postgres -c 'create database twine_api_test'
   - psql -U postgres -c 'create extension postgis'
+
 script:
-  - for x in "api" "dashboard-app" "visitor-app" "lib/twine-util"; do cd $x; npm run lint; cd ..; done;
+  - cd $TRAVIS_BUILD_DIR/$APP_DIR && npm run lint --if-present
   # complile ts/react
-  - for x in "api" "dashboard-app" "visitor-app"; do cd $x; npm run build; cd ..; done;
+  - cd $TRAVIS_BUILD_DIR/$APP_DIR && npm run build --if-present
   # run built server
-  - for x in "api"; do cd $x; npm run checks; cd ..; done;
+  - cd $TRAVIS_BUILD_DIR/$APP_DIR && npm run checks --if-present
   # run tests and coverage
-  - for x in "api" "dashboard-app" "visitor-app" "lib/twine-util"; do cd $x; npm run cover; cd ..; done;
+  - cd $TRAVIS_BUILD_DIR/$APP_DIR && npm run cover --if-present
 
 after_success:
-  - bash $TRAVIS_BUILD_DIR/.travis/cc-report.sh
+  - bash $TRAVIS_BUILD_DIR/.travis/cc_report.sh
+  - aws s3 cp $TRAVIS_BUILD_DIR s3://twine-ci/$TRAVIS_BUILD_NUMBER/ --recursive --exclude "*" --include "coverage-*.json"
+
+jobs:
+  include:
+    - stage: upload
```

Comment:
This job reads as if it only runs on twine-util. Is this correct?

<img width="955" alt="Screen Shot 2019-10-03 at 14 31 02" src="https://user-images.githubusercontent.com/14337958/66130951-7c2de000-e5ea-11e9-84a1-d215f66a3f63.png">

Reply:
Apparently this is how you combine matrix expansion with specific individual jobs. IDK why it picks up that `APP_DIR` value, but the upload job doesn't depend on the `APP_DIR` env var, so it doesn't really matter.